gnc/navigator_perception/helper/model.cc

```diff
@@ -1,11 +1,169 @@
 #include <navigator_vision_lib/model.h>
 
-PerceptionModel::PerceptionModel(float width, float height)
-    : width(width),
-      height(height)
+
+PerceptionModel::PerceptionModel(float width, float height, int min_points): width(width), height(height), min_points(min_points)
 {
+    unused_distances.push_back(width);
+    unused_distances.push_back(height);
+    unused_distances.push_back(sqrt(pow(width, 2) + pow(height, 2)));
 }
 
 PerceptionModel::~PerceptionModel()
 {
 }
+
+
+void PerceptionModel::clear(){
+  potential_models.clear();
+}
+
+bool PerceptionModel::check_point(Eigen::Vector3d point, cv:: Mat img, cv::Matx34d left_cam_mat, bool check)
+
+{
+    current_points.push_back(point);
+    if(current_points.size()  == 1)
+    {
+        return true;
+    }
+    Eigen::Vector3d starting_point = current_points[0];
+    Eigen::Vector3d diff = starting_point - point;
+    float dist_from_starting = sqrt(pow(diff[0], 2) + pow(diff[1], 2) + pow(diff[2], 2));
+    bool ans = false;
+    double mindist = 100;
+    double minel = -1;
+    for(float dist : unused_distances)
+    {
+        float err = dist * .4;
+        float upper = dist + err;
+        float lower = dist - err;
+
+        if(dist_from_starting > lower && dist_from_starting < upper)
+        {
+
+            double m = fabs(dist_from_starting);
+            if(m < mindist){
+              minel = dist;
+              mindist = m;
+            }
+
+            std::stringstream ss;
+            ss<<point(0)<<point(1)<<point(2);
+            point_to_distance.insert(std::pair<std::string,float>(ss.str(), dist));
+            if(current_points.size()  == min_points)
+            {
+                potential_models.push_back(current_points);
+            }
+            ans = true;
+        }
+    }
+    if(ans){
+       unused_distances.erase(std::remove(unused_distances.begin(), unused_distances.end(), minel), unused_distances.end());
```

Comment:
`std::remove` returns an iterator to the address after the last element not removed, which should be the same as `unused_distances.end()`. This would mean that the call to `erase` is not really doing anything here.

Am I missing something here?